src/main/java/hlf/java/rest/client/service/impl/TransactionFulfillmentImpl.java

```diff
@@ -466,4 +471,96 @@ public ResponseEntity<EventAPIResponseModel> readTransactionEventByBlockNumber(
+
+    Collection<Peer> endorsingPeers = new ArrayList<>();
+    String resultString;
+
+    try {
+
+      Network network = gateway.getNetwork(channelName);
+      Contract contract = network.getContract(chaincodeName);
+      Transaction fabricTransaction = contract.createTransaction(transactionFunctionName);
+      // override default commithandler to wait for any response from msp
+      fabricTransaction.setCommitHandler(DefaultCommitHandlers.MSPID_SCOPE_ANYFORTX);
+
+      endorsingPeers =
+          network.getChannel().getPeers().stream()
+              .filter(peer -> multiDataTransactionPayload.getPeerNames().contains(peer.getName()))
+              .collect(Collectors.toList());
+      if (!endorsingPeers.isEmpty()) {
+        fabricTransaction.setEndorsingPeers(endorsingPeers);
+      } else {
+        log.warn("Peer names don't match channel peers");
+      }
+
+      Map<String, byte[]> transientParam = new HashMap<>();
+      List<String> publicParamsList = new ArrayList<>();
+
+      /**
+       * Scan through Private Data details the incoming transaction Request. If the Private data
+       * details consist of a Collection name, then add the key-value pair to the transient map and
+       * also ensure that the Collection name and key name is also part of the public params list.
+       * If Collection name is not present, simply populate the key-value pair to the transient map.
+       */
+      multiDataTransactionPayload
+          .getPrivatePayload()
+          .forEach(
+              privateTransactionPayload -> {
+                if (StringUtils.isNotBlank(privateTransactionPayload.getCollectionName())) {
+                  publicParamsList.add(privateTransactionPayload.getCollectionName());
+                  publicParamsList.add(privateTransactionPayload.getKey());
+                }
+
+                transientParam.put(
+                    privateTransactionPayload.getKey(),
+                    privateTransactionPayload.getData().getBytes(StandardCharsets.UTF_8));
+              });
+
+      fabricTransaction.setTransient(transientParam);
+
+      // Check if Public params are also passed in the request. If provided, append them to the
+      // public params list
+      if (!CollectionUtils.isEmpty(multiDataTransactionPayload.getPublicPayload())) {
+        publicParamsList.addAll(multiDataTransactionPayload.getPublicPayload());
+      }
+
+      // Map to String Array for dispatching via SDK method
+      String[] publicDataArgs =
+          CollectionUtils.isEmpty(multiDataTransactionPayload.getPublicPayload())
+              ? new String[] {}
+              : publicParamsList.toArray(new String[publicParamsList.size()]);
+
+      byte[] result = fabricTransaction.submit(publicDataArgs);
+
+      resultString = new String(result, StandardCharsets.UTF_8);
+      log.info("Transaction Successfully Submitted - Response: " + resultString);
+    } catch (GatewayRuntimeException gre) {
```

Comment:
All exceptions are to be handled like the way it's handled in `writeTransactionToLedger`. The exception `ServiceException` or `FabricTransactionException` decides if the hlf-connector should retry to perform the transaction or not.

Reply:
Sure, this can be done. 
Also, Where are the mentioned Retry mechanism configured for these exceptions? I couldn't find them, will take a second look.

Reply:
Kafka reads are doing the retries, APIs are not recommended for async writes so they are best attempt basis. In case of Kafka reads, the listener is informed about exception for the cases where retry is needed. All other exceptions are silenced through a log trace.

Reply:
Done